navigation_and_routing/lib/src/screens/scaffold_body.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../routing.dart';
+import '../screens/settings.dart';
+import '../widgets/fade_transition_page.dart';
+import 'authors.dart';
+import 'books.dart';
+
+/// Displays the contents of the body of [BookstoreScaffold]
+class BookstoreScaffoldBody extends StatelessWidget {
+  static GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
+
+  const BookstoreScaffoldBody({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var currentRoute = RouteStateScope.of(context)!.route;
+
+    // A nested Router isn't necessary because the back button behavior doesn't
+    // need to be customized.
+    return Navigator(
+      key: navigatorKey,
+      onPopPage: (route, dynamic result) => route.didPop(result),
+      pages: [
+        if (currentRoute.pathTemplate.startsWith('/authors'))
+          const FadeTransitionPage<void>(
+            key: ValueKey('authors'),
+            child: AuthorsScreen(),
+          )
+        else if (currentRoute.pathTemplate.startsWith('/settings'))
+          const FadeTransitionPage<void>(
+            key: ValueKey('settings'),
+            child: SettingsScreen(),
+          )
+        else if (currentRoute.pathTemplate.startsWith('/books') ||
+            currentRoute.pathTemplate == '/')
+          FadeTransitionPage<void>(
+            key: const ValueKey('books'),
+            child: BooksScreen(currentRoute: currentRoute),
+          )
+
+          // Avoid building a Navigator with an empty `pages` list when the
+          // RouteState is set to an unexpected path, such as /signin.
+          //
+          // Since RouteStateScope is an InheritedNotifier, any change to the
```

Comment:
Can you delay the page transition duration to a large number like 1 sec, and hit sign out to see what will happen? I am worrying that the page will suddenly become empty before the page transition finishes

Reply:
Here's a [video](https://drive.google.com/file/d/1i8oBmnFxySWgC7ExuaxWEeJDtj5sg4kq/view?usp=sharing) Yeah this is an issue. 

I think what we need here is  a way to listen for changes to specific routes, rather than just listen to the global `RouteStateScope. This widget only wants to re-build when the route is  to `/authors`, `/settings`, or `/books`.